site/src/scaffold/scripts/popover.ts

```diff
@@ -0,0 +1,72 @@
+export function initPopovers(container: HTMLElement) {
+	container.addEventListener('click', (event: Event) => {
+		let targetPopover: HTMLDetailsElement | null;
+		targetPopover =
+			event.target instanceof Element &&
+			(event.target.closest('details.popover') as HTMLDetailsElement)
+				? (event.target.closest('details.popover') as HTMLDetailsElement)
+				: null;
+
+		if (!targetPopover) {
+			targetPopover =
+				event.target instanceof Element &&
+				event.target.shadowRoot &&
+				event.target.shadowRoot.activeElement &&
+				(event.target.shadowRoot.activeElement.closest('details.popover') as HTMLDetailsElement)
+					? (event.target.shadowRoot.activeElement.closest('details.popover') as HTMLDetailsElement)
+					: null;
+		}
+
+		if (!targetPopover) {
+			return;
+		}
+
+		const allPopovers: HTMLDetailsElement[] = Array.from(
+			container.querySelectorAll('details.popover')
+		);
+
+		if (!targetPopover.open) {
+			const keyHandler = (event: KeyboardEvent) => {
+				if (event.key === 'Escape') {
+					closePopovers();
+				}
+			};
+
+			const clickHandler = (event: Event) => {
```

Comment:
Thinking this should be named something like checkTarget or checkShouldCollapse to describe what it's doing rather than the event it's handling. Plus it's used with focus, touch, and click.

Reply:
Updated